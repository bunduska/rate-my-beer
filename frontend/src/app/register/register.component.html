<div class="image">
    <form [formGroup]="registerForm" (ngSubmit)="onSubmit()">

        <div class="flex-container">

            <div class="h1-content">
                <h1><span class="h1-text">Sign up</span></h1>
            </div>

            <mat-form-field appearance="fill">
                <mat-label>Name</mat-label>
                <mat-icon matPrefix color="primary">person</mat-icon>
                <input class="inputClass" matInput placeholder="Drinker name" formControlName="username" />
            </mat-form-field>
            <mat-error *ngIf="submitted && f.username.errors">Username is required</mat-error>


            <mat-form-field appearance="fill">
                <mat-label>Email</mat-label>
                <mat-icon color="primary" matPrefix>email</mat-icon>
                <input class="inputClass" matInput placeholder="beer@drinking.com" formControlName="email"
                    type="email" />
            </mat-form-field>
            <mat-error *ngIf="submitted && f.email.errors">
                <div *ngIf="f.email.errors.required">Email is required</div>
                <div *ngIf="f.email.errors.email">Email must be a valid email address</div>
            </mat-error>

            <mat-form-field appearance="fill">
                <mat-label>Password</mat-label>
                <mat-icon matPrefix color="primary" class="icon-display"></mat-icon>
                <input class="inputClass" matInput [type]="hide ? 'password' : 'text'" formControlName="password" />
                <a mat-icon-button matSuffix (click)="hide = !hide" [attr.aria-label]="'Hide password'"
                    [attr.aria-pressed]="hide" color="primary" class="hide">
                    <mat-icon>{{ hide ? 'visibility_off' : 'visibility' }}</mat-icon>
                </a>
            </mat-form-field>
            <mat-error *ngIf="submitted && f.password.errors">
                <div *ngIf="f.password.errors.required">Password is required</div>
                <div *ngIf="f.password.errors.minlength">
                    Password must be at least 8 characters
                </div>
            </mat-error>

            <mat-checkbox color="primary" labelPosition="after" formControlName="isLegalAgeConfirmed">
                Please confirm your are old enough to drink in your own country
            </mat-checkbox>
            <mat-error *ngIf="submitted && f.isLegalAgeConfirmed.errors">
                Please confirm!!!
            </mat-error>

            <div class="register-button">
                <button mat-raised-button color="primary">
                    Register
                </button>
            </div>
            <br />
            <div class="button">
                <button mat-raised-button routerLink="/login">Or Login</button>
            </div>
        </div>
    </form>
</div>