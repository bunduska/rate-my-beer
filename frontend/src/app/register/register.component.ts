import { Component, OnInit } from '@angular/core';
import { Router } from '@angular/router';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import { first } from 'rxjs/operators';
import { RegisterService } from '../services/register.service';
import { User } from '../models/user.model';
import { MatSnackBar } from '@angular/material/snack-bar';

@Component({
  selector: 'app-register',
  templateUrl: './register.component.html',
  styleUrls: ['./register.component.css'],
})
export class RegisterComponent implements OnInit {
  registerForm!: FormGroup;
  loading = false;
  submitted = false;
  connectionStatus = false;
  wrongInputStatus = false;
  message = '';
  hide = true;

  constructor(
    private formBuilder: FormBuilder,
    private router: Router,
    private registerService: RegisterService,
    private snackBar: MatSnackBar,
  ) {}

  ngOnInit(): void {
    this.registerForm = this.formBuilder.group({
      email: [
        '',
        [
          Validators.required,
          Validators.email,
          Validators.pattern('^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$'),
        ],
      ],
      username: ['', Validators.required],
      password: ['', [Validators.required, Validators.minLength(8)]],
      isLegalAgeConfirmed: ['', Validators.requiredTrue],
    });
  }

  get f() {
    return this.registerForm.controls;
  }

  onSubmit(): void {
    this.submitted = true;

    if (this.registerForm.invalid) {
      return;
    }

    this.loading = true;
    this.registerService
      .register(this.registerForm.value)
      .pipe(first())
      .subscribe(
        (data: User) => {
          if (data.message !== undefined) {
            this.loading = false;
            this.wrongInputStatus = true;
            this.message = data.message;
            this.openSnackBar(this.message);
          } else {
            this.openSnackBar('Registration is done, please check your e-mail!');
            this.router.navigate(['/login']);
          }
        },
        () => {
          this.loading = false;
          this.connectionStatus = true;
        },
      );
  }

  openSnackBar(message: string) {
    this.snackBar.open(message, 'OK');
  }
}
