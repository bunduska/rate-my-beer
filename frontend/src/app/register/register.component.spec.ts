import { OverlayModule } from '@angular/cdk/overlay';
import { HttpClientModule } from '@angular/common/http';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { MatIconModule } from '@angular/material/icon';
import { MatInputModule } from '@angular/material/input';
import { MatSnackBar } from '@angular/material/snack-bar';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { RegisterComponent } from './register.component';

describe('RegisterComponent', () => {
  let component: RegisterComponent;
  let fixture: ComponentFixture<RegisterComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [RegisterComponent],
      imports: [
        FormsModule,
        ReactiveFormsModule,
        MatCheckboxModule,
        BrowserAnimationsModule,
        OverlayModule,
        MatInputModule,
        MatIconModule,
        RouterTestingModule,
        HttpClientModule,
      ],
      providers: [MatSnackBar],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(RegisterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match with previous snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
