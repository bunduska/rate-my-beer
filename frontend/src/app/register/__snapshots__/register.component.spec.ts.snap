// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterComponent should match with previous snapshot 1`] = `
<app-register
  connectionStatus="false"
  formBuilder={[Function FormBuilder]}
  hide={[Function Boolean]}
  loading="false"
  message=""
  registerForm={[Function FormGroup]}
  registerService={[Function RegisterService]}
  router={[Function Router]}
  snackBar={[Function MatSnackBar]}
  submitted="false"
  wrongInputStatus="false"
>
  <div
    class="image"
  >
    <form
      class="ng-untouched ng-pristine ng-invalid"
      novalidate=""
    >
      <div
        class="flex-container"
      >
        <div
          class="h1-content"
        >
          <h1>
            <span
              class="h1-text"
            >
              Sign up
            </span>
          </h1>
        </div>
        <mat-form-field
          appearance="fill"
          class="mat-form-field mat-primary mat-form-field-type-mat-input mat-form-field-appearance-fill mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              <div
                class="mat-form-field-prefix"
              >
                <mat-icon
                  aria-hidden="true"
                  class="mat-icon notranslate mat-primary material-icons"
                  color="primary"
                  data-mat-icon-type="font"
                  matprefix=""
                  role="img"
                >
                  person
                </mat-icon>
              </div>
              <div
                class="mat-form-field-infix"
              >
                <input
                  aria-invalid="false"
                  aria-required="false"
                  class="inputClass mat-input-element mat-form-field-autofill-control ng-untouched ng-pristine ng-invalid cdk-text-field-autofill-monitored"
                  data-placeholder="Drinker name"
                  formcontrolname="username"
                  matinput=""
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  <label
                    class="mat-form-field-label mat-empty mat-form-field-empty"
                  >
                    <mat-label>
                      Name
                    </mat-label>
                  </label>
                </span>
              </div>
            </div>
            <div
              class="mat-form-field-underline"
            >
              <span
                class="mat-form-field-ripple"
              />
            </div>
            <div
              class="mat-form-field-subscript-wrapper"
            >
              <div
                class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
                style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
              >
                <div
                  class="mat-form-field-hint-spacer"
                />
              </div>
            </div>
          </div>
        </mat-form-field>
        <mat-form-field
          appearance="fill"
          class="mat-form-field mat-primary mat-form-field-type-mat-input mat-form-field-appearance-fill mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              <div
                class="mat-form-field-prefix"
              >
                <mat-icon
                  aria-hidden="true"
                  class="mat-icon notranslate mat-primary material-icons"
                  color="primary"
                  data-mat-icon-type="font"
                  matprefix=""
                  role="img"
                >
                  email
                </mat-icon>
              </div>
              <div
                class="mat-form-field-infix"
              >
                <input
                  aria-invalid="false"
                  aria-required="false"
                  class="inputClass mat-input-element mat-form-field-autofill-control ng-untouched ng-pristine ng-invalid cdk-text-field-autofill-monitored"
                  data-placeholder="beer@drinking.com"
                  formcontrolname="email"
                  matinput=""
                  type="email"
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  <label
                    class="mat-form-field-label mat-empty mat-form-field-empty"
                  >
                    <mat-label>
                      Email
                    </mat-label>
                  </label>
                </span>
              </div>
            </div>
            <div
              class="mat-form-field-underline"
            >
              <span
                class="mat-form-field-ripple"
              />
            </div>
            <div
              class="mat-form-field-subscript-wrapper"
            >
              <div
                class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
                style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
              >
                <div
                  class="mat-form-field-hint-spacer"
                />
              </div>
            </div>
          </div>
        </mat-form-field>
        <mat-form-field
          appearance="fill"
          class="mat-form-field mat-primary mat-form-field-type-mat-input mat-form-field-appearance-fill mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              <div
                class="mat-form-field-prefix"
              >
                <mat-icon
                  aria-hidden="true"
                  class="icon-display mat-icon notranslate mat-primary material-icons"
                  color="primary"
                  data-mat-icon-type="font"
                  matprefix=""
                  role="img"
                />
              </div>
              <div
                class="mat-form-field-infix"
              >
                <input
                  aria-invalid="false"
                  aria-required="false"
                  class="inputClass mat-input-element mat-form-field-autofill-control ng-untouched ng-pristine ng-invalid cdk-text-field-autofill-monitored"
                  formcontrolname="password"
                  matinput=""
                  type="password"
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  <label
                    class="mat-form-field-label mat-empty mat-form-field-empty"
                  >
                    <mat-label>
                      Password
                    </mat-label>
                  </label>
                </span>
              </div>
              <div
                class="mat-form-field-suffix"
              >
                <a
                  aria-label="Hide password"
                  aria-pressed="true"
                  class="hide"
                  color="primary"
                  mat-icon-button=""
                  matsuffix=""
                >
                  <mat-icon
                    aria-hidden="true"
                    class="mat-icon notranslate material-icons mat-icon-no-color"
                    data-mat-icon-type="font"
                    role="img"
                  >
                    visibility_off
                  </mat-icon>
                </a>
              </div>
            </div>
            <div
              class="mat-form-field-underline"
            >
              <span
                class="mat-form-field-ripple"
              />
            </div>
            <div
              class="mat-form-field-subscript-wrapper"
            >
              <div
                class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
                style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
              >
                <div
                  class="mat-form-field-hint-spacer"
                />
              </div>
            </div>
          </div>
        </mat-form-field>
        <mat-checkbox
          class="mat-checkbox mat-primary ng-untouched ng-pristine ng-invalid"
          color="primary"
          formcontrolname="isLegalAgeConfirmed"
          labelposition="after"
        >
          <label
            class="mat-checkbox-layout"
            for="mat-checkbox-2-input"
          >
            <span
              class="mat-checkbox-inner-container"
            >
              <input
                aria-checked="false"
                class="mat-checkbox-input cdk-visually-hidden"
                id="mat-checkbox-2-input"
                tabindex="0"
                type="checkbox"
              />
              <span
                class="mat-checkbox-ripple mat-focus-indicator mat-ripple"
                matripple=""
              >
                <span
                  class="mat-ripple-element mat-checkbox-persistent-ripple"
                />
              </span>
              <span
                class="mat-checkbox-frame"
              />
              <span
                class="mat-checkbox-background"
              >
                <svg
                  aria-hidden="true"
                  class="mat-checkbox-checkmark"
                  focusable="false"
                  version="1.1"
                  viewBox="0 0 24 24"
                  xml:space="preserve"
                >
                  <path
                    class="mat-checkbox-checkmark-path"
                    d="M4.1,12.7 9,17.6 20.3,6.3"
                    fill="none"
                    stroke="white"
                  />
                </svg>
                <span
                  class="mat-checkbox-mixedmark"
                />
              </span>
            </span>
            <span
              class="mat-checkbox-label"
            >
              <span
                style="display:none"
              >
              </span>
              <p>
                 You are old enough to drink in your own country.
              </p>
            </span>
          </label>
        </mat-checkbox>
        <div
          class="register-button"
        >
          <button
            color="primary"
            mat-raised-button=""
          >
            Register
          </button>
        </div>
        <br />
        <div
          class="button"
        >
          <button
            color="accent"
            mat-raised-button=""
            routerlink="/login"
            tabindex="0"
          >
            Or Login
          </button>
        </div>
      </div>
    </form>
  </div>
</app-register>
`;
